ocp_utilities/infra.py

```diff
@@ -79,6 +82,45 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    None of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if one or more nodes condition is other than Ready, i.e
+                                       DiskPressure, MemoryPressure, PIDPressure, NetworkUnavailable
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes = []
+    for node in nodes:
+        node_conditions = node.instance.status.conditions
+        condition_type = []
+        for condition in node_conditions:
+            if (
+                condition.type != Node.Condition.READY
+                and condition.status == Node.Condition.Status.TRUE
+            ):
+                condition_type.append(condition.type)
+
+        if condition_type:
+            condition_type.insert(0, node.name)
+            unhealthy_nodes.append(condition_type)
```

Comment:
Are you inserting these to corrects lists? Seems its the other way around `condition_type` and `nodes` should be inserted in relevant lists

Reply:
Updated: refactored the code for clarity.